cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/ReplicationStatusKey.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.replication;
+
+import co.cask.cdap.api.common.Bytes;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+/**
+ * Handling of Row keys for the Replication State Table
+ */
+public final class ReplicationStatusKey {
+
+  private final String rowType;
+  private final String regionName;
+  private final String rsID;
+  private final byte[] key;
+
+  public ReplicationStatusKey(byte[] rowType, byte[] regionName, byte[] regionServerID) {
+    byte[] rowKey;
+    rowKey = Bytes.add(Bytes.toBytes(rowType.length), rowType);
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(regionName.length), regionName);
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(regionServerID.length), regionServerID);
+    this.rowType = Bytes.toString(rowKey);
+    this.regionName = Bytes.toString(regionName);
+    this.rsID = Bytes.toString(regionServerID);
+    this.key = rowKey;
+  }
+
+  public ReplicationStatusKey(byte[] key) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(key);
+    this.key = key;
+    this.rowType = getString(byteBuffer);
+    this.regionName = getString(byteBuffer);
+    this.rsID = getString(byteBuffer);
+  }
+
+  /**
+   * @return the underlying key
+   */
+  public byte[] getKey() {
+    return key;
+  }
+
+  public String getRowType() {
```

Comment:
It seems that the returned rowType is more often converted back to `byte[]` instead of being used as a String. Maybe it's more efficient to return `byte[]` or have a separate method to return `byte[]`?